### ISSUE TITLE
CommandContext nextMessages() not receiving messages

### ISSUE THREAD

**Issue by PotatoCurry** [l7ssha/nyxx#52]
Whenever I try to use nextMessages(), no messages seem to be received through the stream. Take the following code.
```dart
main() async {
  final bot = NyxxVm(TOKEN);
  CommandsFramework(bot, prefix: "s!")
    ..discoverCommands();

  [other code]
}

@Command("login")
Future<void> login(CommandContext ctx) async {
  print("login start");
  final loginInfo = ctx.nextMessages(3);
  await for (final value in loginInfo) {
    print(value.message.content);
  }
  ctx.reply(content: "abc");

  [other code]
}
```
Ideally, after sending "s!login", this would print "login start" and the next three messages I send the bot in the same channel to the console, then reply with "abc". However, only "login start" is printed, and "abc" is not sent back after sending three messages to the bot.

**Comment by One-Nub** [comment]
Wouldn't you need `await ctx.nextMessages(3)` instead?
and I don't think the await before the for loop is necessary but I may be mistaken

**Comment by PotatoCurry** [comment]
I'm attempting to use the [asynchronous for loop](https://dart.dev/tutorials/language/streams#receiving-stream-events) to iterate over what the stream gives me. As per your suggestion, adding an `await` before `ctx.nextMessages(3)` doesn't change the output.

**Comment by One-Nub** [comment]
Testing it on my own seems to work fine (using the code you provided).
Is it displaying the 3 messages you send? Another thing you could do is add an await before `ctx.reply` (but that shouldn't change anything anyways?)

**Comment by PotatoCurry** [comment]
Yep, pulling the latest from GitHub.

**Comment by PotatoCurry** [comment]
I figured out what's wrong: it doesn't work in direct messages, only in actual text channels in servers.

**Comment by l7ssha** [comment]
DMs are handled the same, it should work, so yeah it isn't indented.

**Comment by l7ssha** [comment]
@PotatoCurry Possibly fixed in a681d0d
